src/open_inwoner/cms/plugins/cms_plugins/tasks.py

```diff
@@ -34,65 +35,98 @@ def render(self, context, instance, placeholder) -> dict[str, str]:
         if not request.user.is_authenticated or not (bsn := request.user.bsn):
             return context
 
-        task_objects = self.get_tasks_by_bsn(user_bsn=bsn)
+        task_objects = self.get_tasks_by_bsn(instance, user_bsn=bsn)
 
         task_dicts = []
         for task in task_objects:
+            if task.soort == "externformulier":
+                task = cast(ExternFormulierTaak, task)
+                base_url = task.portaalformulier.formulier.value
+                param = {"initial_data_reference": task.uuid}
+                task_url = f"{base_url}?{urlencode(param)}"
+            elif task.soort == "url":
+                task = cast(UrlTaak, task)
+                task_url = str(task.task_url.uri)
+            else:
+                task_url = None
```

Comment:
I think it'd be cleaner if we just use `match` here, given we can check the instance. The `soort` still feels a bit implicit to me, let's look directly at the types:


```suggestion
           match task:
               case ExternFormulierTaak():
                base_url = task.portaalformulier.formulier.value
                param = {"initial_data_reference": task.uuid}
                task_url = f"{base_url}?{urlencode(param)}"
              case UrlTaak():
                task_url = str(task.task_url.uri)
            else:
                assert_never(task)
```